src/commands/executeFunction.ts

```diff
@@ -3,35 +3,75 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
+import { HttpMethods } from 'ms-rest';
 import { window } from 'vscode';
 import { SiteClient } from 'vscode-azureappservice';
 import { IActionContext, parseError } from 'vscode-azureextensionui';
 import { ext } from '../extensionVariables';
+import { FuncVersion } from '../FuncVersion';
 import { localize } from '../localize';
 import { FunctionTreeItemBase } from '../tree/FunctionTreeItemBase';
 import { RemoteFunctionTreeItem } from '../tree/remoteProject/RemoteFunctionTreeItem';
+import { nonNullProp } from '../utils/nonNull';
 import { requestUtils } from '../utils/requestUtils';
 
 export async function executeFunction(context: IActionContext, node?: FunctionTreeItemBase): Promise<void> {
     if (!node) {
-        const noItemFoundErrorMessage: string = localize('noTimerFunctions', 'No timer functions found.');
-        node = await ext.tree.showTreeItemPicker<FunctionTreeItemBase>(/Function;Timer;/i, { ...context, noItemFoundErrorMessage });
+        const noItemFoundErrorMessage: string = localize('noFunctions', 'No functions found.');
+        node = await ext.tree.showTreeItemPicker<FunctionTreeItemBase>(/Function;/i, { ...context, noItemFoundErrorMessage });
     }
 
-    const name: string = node.name;
     const client: SiteClient | undefined = node instanceof RemoteFunctionTreeItem ? node.parent.parent.root.client : undefined;
-    const hostUrl: string = node.parent.parent.hostUrl;
+    const triggerBindingType: string | undefined = node.config.triggerBinding?.type;
+    context.telemetry.properties.triggerBindingType = triggerBindingType;
+
+    let functionInput: string | {} = '';
+    if (!node.config.isTimerTrigger) {
+        const prompt: string = localize('enterRequestBody', 'Enter request body');
+        let value: string | undefined;
+        if (triggerBindingType) {
+            const version: FuncVersion = await node.parent.parent.getVersion();
+            value = await ext.templateProvider.tryGetSampleData(context, version, triggerBindingType);
+            if (value) {
+                // Clean up the whitespace to make it more friendly for a one-line input box
+                value = value.replace(/[\r\n\t]/g, ' ');
+                value = value.replace(/ +/g, ' ');
```

Comment:
Does this turn multiple spaces to one?

Reply:
Yep - "sample.dat" file uses multiple spaces for tabs, which looks weird in the input box